webapp/store/reducers/clients.js

```diff
@@ -0,0 +1,32 @@
+import { SET_CLIENTS, SET_DEFAULT_CLIENT } from '../constants/clients';
+import assert from 'bsert';
+
+const initialState = {
+  currentClient: {},
+  clients: {}
+};
+
+const clientsState = (state = initialState, action) => {
+  let newState = { ...state };
+  const { payload, type } = action;
+  switch (type) {
+    case SET_CLIENTS: {
+      newState.clients = payload;
+      return newState;
+    }
+
+    case SET_DEFAULT_CLIENT: {
+      assert(
+        typeof payload === 'object' && payload.id,
+        'Must have a client object with an id'
+      );
+      if (state.clients[payload.id]) newState.currentClient = payload;
```

Comment:
This line seems like it could result in a quiet error

Reply:
Actually, I'm going to take this out. This would stop us from setting some id that's not on the server which I'm not sure we want to do. I feel like this could still result in some weird edge cases, but at least like this, if you use the BPClient to query a node that's not on the proxy server, you'll get an appropriate error.